numcosmo/lss/nc_galaxy_wl.c

```diff
@@ -289,13 +296,83 @@ nc_galaxy_wl_eval_m2lnP (NcGalaxyWL *gwl, NcHICosmo *cosmo, NcHaloDensityProfile
   NcGalaxyWLEval gwleval         = {self->wl_dist, cosmo, dp, smd, z_cluster, 0};
   gdouble res                    = 0.0;
   gint gal_i;
+
+  nc_galaxy_wl_dist_m2lnP_initial_prep (self->wl_dist, self->gz_dist, cosmo, dp, smd, z_cluster);
   
+  // if (self->no_kde)
+  // {
   for (gal_i = 0; gal_i < self->len; gal_i++)
   {
     gwleval.gal_i = gal_i;
-    nc_galaxy_wl_dist_m2lnP_prep (self->wl_dist, cosmo, dp, smd, z_cluster, gal_i);
+    // nc_galaxy_wl_dist_m2lnP_prep (self->wl_dist, cosmo, dp, smd, z_cluster, gal_i);
```

Comment:
Why did you removed this? It will break the code for objects that require preparation for each galaxy.